clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerInterceptors.java

```diff
@@ -123,6 +122,10 @@ public void onSendError(ProducerRecord<K, V> record, TopicPartition interceptTop
         }
     }
 
+    public static <K, V> TopicPartition createTopicPartition(ProducerRecord<K, V> record) {
```

Comment:
This is not a very good name since "creating a partition" usually means calling `createPartitions`. How about `extractTopicPartition` ?